cfme/cloud/instance/openstack.py

```diff
@@ -54,6 +54,44 @@ def ui_powerstates_unavailable(self):
             'on': [self.START],
             'off': [self.SUSPEND, self.SOFT_REBOOT, self.HARD_REBOOT]}
 
+    def attach_volume(self, name, mountpoint=None):
+        initial_volume_count = self.volume_count
+        view = navigate_to(self, 'AttachVolume')
+        view.form.volume.select_by_visible_text(name)
+        if mountpoint:
+            view.form.mountpoint.fill(mountpoint)
+        view.form.submit_button.click()
+        view.flash.assert_success_message(
+            'Attaching Cloud Volume "{}" to {} finished'.format(name, self.name))
+
+        wait_for(lambda: self.volume_count > initial_volume_count,
+                 delay=20,
+                 timeout=300,
+                 fail_func=self.refresh_relationships())
+
+    def detach_volume(self, name):
+        initial_volume_count = self.volume_count
+        view = navigate_to(self, 'DetachVolume')
+        view.form.volume.select_by_visible_text(name)
```

Comment:
Required: view.form.volume.fill(name)

Reply:
i tried with fill before and it did not work
Seems because it is a dropdown list, not textbox

Reply:
volume is BootstrapSelect and has fill method that uses also select_by_visible_text
https://github.com/RedHatQE/widgetastic.patternfly/blob/fdab963a7b34e7d10f32cc198fadb9b9eb7bfe17/src/widgetastic_patternfly/__init__.py#L986

Reply:
Thanks for clarification,  will try again